contracts/interchain-token-service/src/state.rs

```diff
@@ -25,6 +27,7 @@ pub struct Config {
 
 const CONFIG: Item<Config> = Item::new("config");
 const ITS_CONTRACTS: Map<&ChainNameRaw, Address> = Map::new("its_contracts");
+const GATEWAY_TOKEN_DENOMS: Map<TokenId, String> = Map::new("gateway_token_denoms");
```

Comment:
add a query to get the map? pagination isn't necessary since gateway token count is bounded (<100)